frontend/src/console/components/InstanceSettings/InstanceSettings.tsx

```diff
@@ -72,17 +94,38 @@ export class InstanceSettingsComponent extends React.PureComponent<
     const field = e.target.name;
     const { value } = e.target;
 
-    this.setState({
-      [field]: value
-    });
+    if (field === 'instanceName') {
+      this.setState({
+        instanceName: value
+      });
+    }
+    if (field === 'publicContactEmail') {
+      this.setState({
+        publicContactEmail: value
+      });
+    }
   };
 
   private updateValue = (fieldName: string, value: string) => {
     const instance = this.props.activeInstance;
 
-    // Only make update request if field changed
-    if (instance.data && this.state[fieldName] !== instance.data[fieldName]) {
-      this.props.updateFieldValue(instance.data.id, fieldName, value);
+    // Only make update request if instance name changed
+    if (
+      instance.data &&
+      this.state.instanceName !== instance.data.instanceName
+    ) {
+      this.props.updateFieldValue(instance.data.id, 'instanceName', value);
+    }
+    // Only make update request if public contact email changed
+    if (
+      instance.data &&
+      this.state.publicContactEmail !== instance.data.publicContactEmail
+    ) {
+      this.props.updateFieldValue(
+        instance.data.id,
+        'publicContactEmail',
+        value
+      );
```

Comment:
Did you rewrite this because it was saying that the `string` cannot be used for indexing `State`?

If so, this should work:
```ts
if (instance.data && this.state[fieldName as keyof State] !== instance.data[fieldName]) {
  this.props.updateFieldValue(instance.data.id, fieldName, value);
}
```

Reply:
I was confused about how to fix it, and I don't know that I can use it this way, thanks
I changed it :)